tests/test_pipenv.py

```diff
@@ -590,6 +590,55 @@ def test_sequential_mode(self, pypi):
             c = p.pipenv('run python -c "import requests; import idna; import certifi; import records; import tpfd; import parse;"')
             assert c.return_code == 0
 
+    @pytest.mark.install
+    @pytest.mark.run
+    def test_pep423_name_install(self, pypi):
+        with PipenvInstance(pypi=pypi) as p:
+            with open(p.pipfile_path, 'w') as f:
+                contents = """
+# Pre comment
+[packages]
+python_DateUtil = "*"   # Inline comment
+"""
+                f.write(contents)
+
+            c = p.pipenv('install')
+            assert c.return_code == 0
+
+            c = p.pipenv('install requests')
+            assert c.return_code == 0
+            assert 'python_DateUtil' in p.pipfile['packages']
+            c = p.pipenv('install python_dateutil')
+            assert c.return_code == 0
+            assert 'python-dateutil' in p.pipfile['packages']
+            assert 'python_DateUtil' not in p.pipfile['packages']
+            contents = open(p.pipfile_path).read()
+            assert '# Pre comment' in contents
+            assert '# Inline comment' in contents
```

Comment:
Another couple of test cases are needed to check that:

1. the version constraint is preserved for the `python_DateUtil = "==1.5"` case if no version constraint is given on the command line

2. the version constraint is replaced if an explicit version constraint is given on the command line (including resetting it to a wildcard dependency with `pipenv install 'python_DateUtil=*'`)